ocaml/tests/test_sdn_controller.ml

```diff
@@ -12,120 +12,132 @@
-  assert_equal address (Db.SDN_controller.get_address ~__context ~self:sdn_controller);
-  assert_equal port (Db.SDN_controller.get_port ~__context ~self:sdn_controller)
+  Alcotest.check (Alcotest_comparators.from_rpc_of_t API.rpc_of_sdn_controller_protocol)
+    "test_sdn_controller_introduce_ok: protocol"
+    protocol
+    (Db.SDN_controller.get_protocol ~__context ~self:sdn_controller);
+  Alcotest.(check string)
+    "test_sdn_controller_introduce_ok: address"
+    address
+    (Db.SDN_controller.get_address ~__context ~self:sdn_controller);
+  Alcotest.(check int64)
+    "test_sdn_controller_introduce_ok: port"
+    port
+    (Db.SDN_controller.get_port ~__context ~self:sdn_controller)
 
 let test_sdn_controller_pssl_invalid_address () =
   let protocol = `pssl in
   let address = "192.168.1.10" in
   let port = 0L in
-  let __context = make_test_database () in
-  assert_raises_api_error
-    Api_errors.invalid_value
-    ~args:["address"; "192.168.1.10"]
-    (fun () -> Xapi_sdn_controller.db_introduce ~__context ~protocol ~address ~port)
+  let __context = T.make_test_database () in
+  Alcotest.check_raises "test_sdn_controller_pssl_invalid_address"
+    Api_errors.(Server_error (invalid_value, ["address"; "192.168.1.10"]))
+    (fun () -> Xapi_sdn_controller.db_introduce ~__context ~protocol ~address ~port |> ignore)
 
 let test_sdn_controller_pssl_invalid_port () =
   let protocol = `pssl in
   let address = "" in
   let port = 1L in
-  let __context = make_test_database () in
-  assert_raises_api_error
-    Api_errors.invalid_value
-    ~args:["port"; "1"]
-    (fun () -> Xapi_sdn_controller.db_introduce ~__context ~protocol ~address ~port)
+  let __context = T.make_test_database () in
+  Alcotest.check_raises "test_sdn_controller_pssl_invalid_port"
+    Api_errors.(Server_error (invalid_value, ["port"; "1"]))
+    (fun () -> Xapi_sdn_controller.db_introduce ~__context ~protocol ~address ~port |> ignore)
 
 let test_sdn_controller_ssl_default_port () =
   let protocol = `ssl in
   let address = "192.168.1.10" in
   let port = 0L in
-  let __context = make_test_database () in
+  let __context = T.make_test_database () in
   let sdn_controller = Xapi_sdn_controller.db_introduce ~__context ~protocol ~address ~port in
-  assert_equal protocol (Db.SDN_controller.get_protocol ~__context ~self:sdn_controller);
-  assert_equal address (Db.SDN_controller.get_address ~__context ~self:sdn_controller);
-  assert_equal 6632L (Db.SDN_controller.get_port ~__context ~self:sdn_controller)
+  Alcotest.check (Alcotest_comparators.from_rpc_of_t API.rpc_of_sdn_controller_protocol)
+    "test_sdn_controller_ssl_default_port: protocol"
+    protocol
+    (Db.SDN_controller.get_protocol ~__context ~self:sdn_controller);
+  Alcotest.(check string)
+    "test_sdn_controller_ssl_default_port: address"
+    address
+    (Db.SDN_controller.get_address ~__context ~self:sdn_controller);
+  Alcotest.(check int64)
+    "test_sdn_controller_ssl_default_port: port"
```

Comment:
There is no need to prefix these messages with the test name as Alcotest will say which test failed :)

Reply:
It does when outputting all of the test results, but I thought it might be handy to name the individual test within the test log itself, which doesn't include the title.